addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/AbstractHostFilePlugin.java

```diff
@@ -0,0 +1,212 @@
+    }
+
+    private String getOtherInfo() {
+        return Constant.messages.getString(messagePrefix + "otherinfo");
+    }
+
+    /**
+     * Always returns true - override to add functionality to match specific content
+     *
+     * @param msg the message being scanned (after being sent).
+     * @return true if the content matches
+     */
+    public boolean hasContent(HttpMessage msg) {
+        return true;
+    }
+
+    /**
+     * Always returns false - override to add functionality to detect FPs
+     *
+     * @param msg the message being scanned (after being sent).
+     * @return true if its a false positive
+     */
+    public boolean isFalsePositive(HttpMessage msg) {
+        return false;
+    }
+
+    @Override
+    public void scan() {
+
+        // Check if the user stopped things. One request per URL so check before
+        // sending the request
+        if (isStop()) {
+            LOG.debug("Scan rule {} Stopping.", getName());
+            return;
+        }
+
+        HttpMessage newRequest = getNewMsg();
+        newRequest.getRequestHeader().setMethod(HttpRequestHeader.GET);
+        newRequest.getRequestHeader().setHeader(HttpHeader.CONTENT_TYPE, null);
+        newRequest.setRequestBody("");
+        URI baseUri = getBaseMsg().getRequestHeader().getURI();
+        URI fileUri = null;
+        try {
+            fileUri =
+                    new URI(
+                            baseUri.getScheme(),
+                            null,
+                            baseUri.getHost(),
+                            baseUri.getPort(),
+                            getFilename());
+        } catch (URIException uEx) {
+            LOG.debug(
+                    "An error occurred creating a URI for the: {} rule. {}",
+                    getName(),
+                    uEx.getMessage(),
+                    uEx);
+            return;
+        }
+        try {
+            newRequest.getRequestHeader().setURI(fileUri);
+        } catch (URIException uEx) {
+            LOG.debug(
+                    "An error occurred setting the URI for a new request used by: {} rule. {}",
+                    getName(),
+                    uEx.getMessage(),
+                    uEx);
+            return;
+        }
+        try {
+            sendAndReceive(newRequest, false);
+        } catch (IOException e) {
+            LOG.warn(
+                    "An error occurred while checking [{}] [{}] for {} Caught {} {}",
+                    newRequest.getRequestHeader().getMethod(),
+                    newRequest.getRequestHeader().getURI(),
+                    getName(),
+                    e.getClass().getName(),
+                    e.getMessage());
+            return;
+        }
+        if (isFalsePositive(newRequest)) {
+            return;
+        }
+        int statusCode = newRequest.getResponseHeader().getStatusCode();
+        if (statusCode == HttpStatusCode.OK || isPage200(newRequest)) {
```

Comment:
Explicitly checking the status code defeats the custom page definition, maybe use just `isSuccess`?